src/test/java/api/loans/anonymization/AnonymizeLoansImmediatelyAPITests.java

```diff
@@ -0,0 +1,223 @@
+  }
+
+  /**
+   * Scenario 2
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Immediately after fee/fine closes"
+   *         A closed loan with an associated fee/fine
+   *     When all fees/fines associated with the loan are closed
+   *     Then anonymize the loan
+   */
+  @Test
+  public void testClosedLoansWithClosedFeesAndIsAnonymized()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    LoanHistoryConfigurationBuilder loanHistoryConfig = new LoanHistoryConfigurationBuilder()
+      .loanCloseAnonymizeImmediately()
+      .feeFineCloseAnonymizeImmediately();
+    createConfiguration(loanHistoryConfig);
+
+    IndividualResource loanResource = loansFixture.checkOutByBarcode(new CheckOutByBarcodeRequestBuilder().forItem(item1)
+      .to(user)
+      .at(servicePoint.getId()));
+    UUID loanID = loanResource.getId();
+
+    createClosedAccountWithFeeFines(loanResource, DateTime.now());
+
+    loansFixture.checkInByBarcode(item1);
+
+    anonymizeLoansInTenant();
+
+    assertThat(loansStorageClient.getById(loanID)
+      .getJson(), isAnonymized());
+  }
+
+  /**
+   * Scenario 3
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Immediately after fee/fine closes"
+   *         A closed loan with an associated fee/fine
+   *     When all fees/fines associated with the loan are closed
+   *     Then anonymize the loan
+   */
+  @Test
+  public void testClosedLoansWithOpenFeesAndFinesIsNotAnonymized()
+    throws InterruptedException, ExecutionException, TimeoutException, MalformedURLException {
+
+    LoanHistoryConfigurationBuilder loanHistoryConfig = new LoanHistoryConfigurationBuilder()
+      .loanCloseAnonymizeImmediately()
+      .feeFineCloseAnonymizeNever();
+    createConfiguration(loanHistoryConfig);
+
+    IndividualResource loanResource = loansFixture.checkOutByBarcode(new CheckOutByBarcodeRequestBuilder().forItem(item1)
+      .to(user)
+      .at(servicePoint.getId()));
+    UUID loanID = loanResource.getId();
+
+    createOpenAccountWithFeeFines(loanResource);
+
+    loansFixture.checkInByBarcode(item1);
+
+    anonymizeLoansInTenant();
+
+    assertThat(loansStorageClient.getById(loanID).getJson(),
+      not(isAnonymized()));
+  }
+
+  /**
+   * Scenario 4
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Never"
+   *         An open loan with an associated fee/fine
+   *     When all fees/fines associated with the loan are closed
+   *     Then do not anonymize the loan
+   */
+  @Test
+  public void testClosedLoansWithClosedFeesAndFinesIsNotAnonymized()
```

Comment:
The scenario describes how an open loan with an open fees / fines account is not anonymized. However the name and definition suggests the fees / fines account is closed. Which is correct?

Reply:
Tests have been updated